docs/reference/builder.md

```diff
@@ -908,6 +909,7 @@ Each `<src>` may contain wildcards and matching will be done using Go's
 
     COPY hom* /mydir/        # adds all files starting with "hom"
     COPY hom?.txt /mydir/    # ? is replaced with any single character, e.g., "home.txt"
+    COPY arr[[]0].txt /mydir/    # file name will be "arr[0].txt"
```

Comment:
I wonder if we should put this example separate from the other; in this case we're not really doing "matching", but try to _avoid_ Golang seeing it as a matching rule.

```Dockerfile
When copying files or directories that contain special characters (such as `[`
and `]`), you need to escape those paths following the Golang rules to prevent
them from being treated as a matching pattern. For example, to copy a file
named `arr[0].txt`, use the following;

    COPY arr[[]0].txt /mydir/    # copy a file named "arr[0].txt" to /mydir/

```

Are there other characters that need escaping that we know of?